libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialogBot.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Generators;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Resources;
+using Microsoft.Bot.Builder.Skills;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive
+{
+    /// <summary>
+    /// An <see cref="IBot"/> implementation that hosts an <see cref="AdaptiveDialog"/>.
+    /// </summary>
+    public class AdaptiveDialogBot : IBot
+    {
+        private const string DefaultLocale = "en-US";
+        private const string DefaultLg = "main.lg";
+
+        private static ConcurrentDictionary<ResourceExplorer, LanguageGeneratorManager> _languageGeneratorManagers = new ConcurrentDictionary<ResourceExplorer, LanguageGeneratorManager>();
+
+        private readonly string _defaultLocale;
+        private readonly string _rootDialogId;
+
+        private readonly ILogger _logger;
+        private readonly ResourceExplorer _resourceExplorer;
+        private readonly IStorage _storage;
+        private readonly BotFrameworkClient _botFrameworkClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AdaptiveDialogBot"/> class.
+        /// </summary>
+        /// <param name="resourceExplorer">The Bot Builder <see cref="ResourceExplorer"/> to load the <see cref="Dialog"/> from.</param>
+        /// <param name="rootDialogId">The id of the dialog to load from the <see cref="ResourceExplorer"/>.</param>
+        /// <param name="defaultLocale">The default locale for this bot.</param>
+        /// <param name="logger">An <see cref="ILogger"/> instance.</param>
+        /// <param name="storage">The <see cref="IStorage"/> implementation to use for this <see cref="Dialog"/>.</param>
+        /// <param name="botFrameworkClient">A <see cref="BotFrameworkClient"/> for making calls to Bot Builder Skills.</param>
+        public AdaptiveDialogBot(
+            ResourceExplorer resourceExplorer,
+            string rootDialogId,
+            string defaultLocale,
+            ILogger<AdaptiveDialogBot> logger = null,
+            IStorage storage = null,
+            BotFrameworkClient botFrameworkClient = null)
+        {
+            _resourceExplorer = resourceExplorer;
+            _rootDialogId = rootDialogId ?? "Main.dialog";
```

Comment:
define constant like you did for default locale?

Reply:
done